platform/ui/src/components/InputDoubleRange/InputDoubleRange.tsx

```diff
@@ -0,0 +1,219 @@
+import React, { useState, useCallback, useEffect } from 'react';
+import classNames from 'classnames';
+import { InputNumber } from '../../components'; // Import InputNumber component
+import './InputDoubleRange.css';
+
+type InputDoubleRangeProps = {
+  values: [number, number];
+  onChange: (values: [number, number]) => void;
+  minValue?: number;
+  maxValue?: number;
+  step?: number;
+  unit?: string;
+  containerClassName?: string;
+  inputClassName?: string;
+  labelClassName?: string;
+  labelVariant?: string;
+  showLabel?: boolean;
+  labelPosition?: 'left' | 'right';
+  trackColor?: string;
+  allowNumberEdit?: boolean;
+  showAdjustmentArrows?: boolean;
+};
+
+const InputDoubleRange: React.FC<InputDoubleRangeProps> = ({
+  values,
+  onChange,
+  minValue,
+  maxValue,
+  step,
+  unit,
+  containerClassName,
+  inputClassName,
+  labelClassName,
+  labelVariant,
+  showLabel,
+  labelPosition,
+  trackColor,
+  allowNumberEdit,
+  showAdjustmentArrows,
+}) => {
+  // Set initial thumb positions as percentages
+  const initialPercentageStart = Math.round(((values[0] - minValue) / (maxValue - minValue)) * 100);
+  const initialPercentageEnd = Math.round(((values[1] - minValue) / (maxValue - minValue)) * 100);
+  const [percentageStart, setPercentageStart] = useState(initialPercentageStart);
+  const [percentageEnd, setPercentageEnd] = useState(initialPercentageEnd);
+
+  const [rangeValue, setRangeValue] = useState(values);
+
+  const updateRangeValues = (newValues, index = null) => {
+    const updatedRangeValue = Array.isArray(newValues) ? [...newValues] : [...rangeValue];
+    if (index !== null) {
+      updatedRangeValue[index] = newValues;
+    }
+
+    const calculatePercentage = value => ((value - minValue) / (maxValue - minValue)) * 100;
+
+    const newPercentageStart = calculatePercentage(updatedRangeValue[0]);
+    const newPercentageEnd = calculatePercentage(updatedRangeValue[1]);
+
+    setRangeValue(updatedRangeValue);
+    onChange(updatedRangeValue);
+
+    setPercentageStart(newPercentageStart);
+    setPercentageEnd(newPercentageEnd);
+  };
+
+  useEffect(() => {
+    updateRangeValues(values);
+  }, [values, minValue, maxValue]);
+
+  const LabelOrEditableNumber = (val, index) => {
+    return allowNumberEdit ? (
+      <InputNumber
+        minValue={minValue}
+        maxValue={maxValue}
+        value={val}
+        onChange={newValue => {
+          updateRangeValues(newValue, index);
+        }}
+        step={step}
+        labelClassName="text-white"
+        showAdjustmentArrows={showAdjustmentArrows}
+      />
+    ) : (
+      <span className={classNames(labelClassName ?? 'text-white')}>
+        {val}
+        {unit}
+      </span>
+    );
+  };
+
+  useEffect(() => {
+    return () => {
+      window.removeEventListener('mousemove', handleMouseMove);
```

Comment:
I might be wrong but I think we might have a slight memory leak here - ableit unlikely to occur. During a drag of either thumb, `InputDoubleRange` is rerendered constantly and as such a "new" `handleGlobalMouseUp` and `handleMouseMove` functions are created for each render. When these `removeEventListener` functions are called, the referenced `handleGlobalMouseUp` and `handleMouseMove` are NOT the ones that were added in `handleMouseDown`. However this would only be a problem if the component were unmounted during a drag. So maybe we don't need to worry about it too much, but I think we can solve the problem by simply using `useCallback` for each listner function.

Reply:
Thanks for noticing this, but as you said let's worry when we have a bug for this